src/Compilers/CSharp/Portable/Symbols/MergedNamespaceSymbol.cs

```diff
@@ -168,17 +168,42 @@ private ImmutableArray<Symbol> SlowGetChildrenOfName(ReadOnlyMemory<char> name)
         /// </summary>
         private HashSet<ReadOnlyMemory<char>> SlowGetChildNames(IEqualityComparer<ReadOnlyMemory<char>> comparer)
         {
-            var childNames = new HashSet<ReadOnlyMemory<char>>(comparer);
+            // compute the final capacity of the set we'll return, to reduce heap churn
+            int childCount = 0;
+            List<ImmutableArray<Symbol>> namespacesMembers = new(capacity: _namespacesToMerge.Length);
 
             foreach (var ns in _namespacesToMerge)
             {
-                foreach (var child in ns.GetMembersUnordered())
+                var members = ns.GetMembersUnordered();
+                childCount += members.Length;
+
+                namespacesMembers.Add(members);
+            }
+
+#if NETSTANDARD2_0
+            // HashSet<T> in netstandard2.0 doesn't have a constructor that accepts a capacity
+            // parameter.  It does, however, optimally allocate for the appropriate capacity
+            // if it's created with an IEnumerable<T> that also implements ICollection<T>.  We'll
+            // reduce overall heap churn by first creating a list of elements to go into the
+            // HashSet and then initialzing the HashSet from the list.
+            List<ReadOnlyMemory<char>> childNames = new(childCount);
+#else
+            HashSet<ReadOnlyMemory<char>> childNames = new(childCount, comparer);
+#endif
+
+            foreach (var nsMembers in namespacesMembers)
```

Comment:
>namespacesMembers [](http://example.com/codeflow?start=38&length=17)

Is there a real advantage from building this list? Could we simply iterate through ```_namespacesToMerge``` here and only calculate the count in the first loop?

Reply:
I built the list to avoid calling `GetMembersUnordered` multiple times for each namespace, assuming that I couldn't count on it to be cheap.  I couldn't easily tell by browsing `NamespaceOrTypeSymbol` because `GetMembersUnordered` is abstract.

If that's not a good assumption (i.e. I don't really need to be worried about the runtime or allocation overhead of `GetMembersUnordered`), I agree that the extra complexity of building the list is unnecessary.

Reply:
yeah, once you call it once, it will be computed/cached.  so since we have to pay the price no matter what, just call it to get hte count, then you can use that final count and call again to get back the members.  They will not have changed (or else the world would blow up in different ways :)).

Reply:
Good call.  Simplifying is goodness.